### PR TITLE
Ticket6

### PR DESCRIPTION
**Pull request by bigsegs**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -114,8 +114,50 @@ describe("GET /api/topics ", () => {
        
         
         })
-    });
+      })
 
-    
+    })
+
```
Comment (pippaamy) [inline __tests__/app.test.js:120]: we shouldn't be pushing up with .only!!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -114,8 +114,50 @@ describe("GET /api/topics ", () => {
...
+        expect(result.body.msg).toEqual("Invalid article Id")
+      })
+    })
+
+    it('should return 200 with correct properties',()=>{
+      return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then((result)=>{
+        expect(result.body.comments).toBeSortedBy('created_at',{descending:true})
```
Comment (pippaamy) [inline __tests__/app.test.js:136]: check the length before the forEach

File: __tests__/app.test.js [hunk 3]
```diff
@@ -114,8 +114,50 @@ describe("GET /api/topics ", () => {
...
+        expect(comment).toHaveProperty('author');
+        expect(comment).toHaveProperty('body');
+        expect(comment).toHaveProperty('article_id');
+        expect(typeof(comment.comment_id)).toBe('number');
+        expect(typeof(comment.votes)).toBe('number');
+        expect(typeof(comment.created_at)).toBe('string');
+        expect(typeof(comment.author)).toBe('string');
+        expect(typeof(comment.body)).toBe('string');
+        expect(typeof(comment.article_id)).toBe('number');
+        })
```
Comment (pippaamy) [inline __tests__/app.test.js:151]: can we add another 200 test for a valid id but with no comments

File: commentController.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+const article_id=req.params.article_id;
+
+selectCommentsByArticleId(article_id).then((result)=>{
+    
+    res.status(200).send({comments:result})
+})
+.catch((err)=>{
+   
+res.status(err.status).send({msg:err.msg})
```
Comment (pippaamy) [inline commentController.js]: we should have this error handling in a separate file

**Review by pippaamy — COMMENTED** [review]
good job, some changes to make and then pop in another pr for this one !! :)

File: app.js [hunk 5]
```diff
@@ -15,6 +15,11 @@ app.get('/api/articles',getAllArticles);
 
 app.get('/api/articles/:article_id/comments',getCommentsByArticleId);
 
+app.use((err,req,res,next)=>{
+    
+    res.status(err.status).send({msg:err.msg})
+})
```
Comment (xViolaine) [inline app.js:21]: Unrelated to the ticket, but it would be good to include an error handler for a 500 Internal Server Error!

**Review by xViolaine — APPROVED** [review]

**Pull request merged**